compiler/rustc_privacy/src/lib.rs

```diff
@@ -1808,50 +1793,6 @@ impl SearchInterfaceForPrivateItemsVisitor<'_> {
         let hir_id = self.tcx.hir().local_def_id_to_hir_id(local_def_id);
         let span = self.tcx.def_span(self.item_def_id.to_def_id());
         let vis_span = self.tcx.def_span(def_id);
-        if !vis.is_at_least(self.required_visibility, self.tcx) {
-            let vis_descr = match vis {
-                ty::Visibility::Public => "public",
-                ty::Visibility::Restricted(vis_def_id) => {
-                    if vis_def_id == self.tcx.parent_module(hir_id) {
-                        "private"
-                    } else if vis_def_id.is_top_level_module() {
-                        "crate-private"
-                    } else {
-                        "restricted"
-                    }
-                }
-            };
-
-            if self.has_old_errors
-                || self.in_assoc_ty
```

Comment:
If `in_assoc_ty` is true we need to report hard errors based on nominal visibilities in the same way as previously.
https://github.com/rust-lang/rfcs/blob/master/text/2145-type-privacy.md#additional-restrictions-for-associated-items